primefaces/src/main/java/org/primefaces/component/feedreader/FeedInput.java

```diff
@@ -23,34 +23,86 @@
  */
 package org.primefaces.component.feedreader;
 
-import java.net.URL;
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
-import com.rometools.rome.feed.synd.SyndFeed;
-import com.rometools.rome.io.FeedException;
-import com.rometools.rome.io.SyndFeedInput;
-import com.rometools.rome.io.XmlReader;
-import java.io.IOException;
+import org.primefaces.model.feedreader.FeedItem;
+import org.primefaces.util.Constants;
+
+import com.apptasticsoftware.rssreader.AbstractRssReader;
+import com.apptasticsoftware.rssreader.Item;
+import com.apptasticsoftware.rssreader.RssReader;
+import com.apptasticsoftware.rssreader.module.itunes.ItunesItem;
+import com.apptasticsoftware.rssreader.module.itunes.ItunesRssReader;
 
+/**
+ * Utility class to parse feed input into a list of FeedItem objects.
+ */
 public class FeedInput {
```

Comment:
Utility class, then:

- final?
- private ctor?
- static public methods ?

Reply:
Rename into RSSUtlis ? (FeedInput sounds very much like a POJO)

Reply:
That's how PrimeTek had it before as FeedInput I think to make sure the classes for the library are not in the Renderer causing any classloading issues.  I will clean it up though as it is a bad name.

Reply:
Not sure I get that point

> I think to make sure the classes for the library are not in the Renderer causing any classloading issues

Even if they are present in the renderer, it shouldn't be an issue

Reply:
I am not a classloader expert so @tandraschko @jepsar can correct me but I know in some servers like Payara what happens is the `faces-config.xml` gets scanned at startup and all `Renderer` classes are classloaded.  If you use a class in a Renderer that is NOT present the startup fails.  So if you look at our Optional dependencies here https://primefaces.github.io/primefaces/14_0_0/#/gettingstarted/dependencies?id=dependencies none of those are actually used in a Renderer they are all loaded from a separate class.

So I assume that was PrimeTek's reasoning behind this `FeedInput` class being separate as the only place the `com.rometools` was imported.  

This is ALL just a guess based on past errors i have seen in Payara etc.?

Reply:
Refactored into static `RSSUtils.java`

Reply:
> I am not a classloader expert so @tandraschko @jepsar can correct me but I know in some servers like Payara what happens is the `faces-config.xml` gets scanned at startup and all `Renderer` classes are classloaded. If you use a class in a Renderer that is NOT present the startup fails. So if you look at our Optional dependencies here https://primefaces.github.io/primefaces/14_0_0/#/gettingstarted/dependencies?id=dependencies none of those are actually used in a Renderer they are all loaded from a separate class.
> 
> So I assume that was PrimeTek's reasoning behind this `FeedInput` class being separate as the only place the `com.rometools` was imported.
> 
> This is ALL just a guess based on past errors i have seen in Payara etc.?

Not sure again what you're replying to 😅 (sorry...), but anyway RSSUtils looks fine

Reply:
I was replying to your comment "Even if they are present in the renderer, it shouldn't be an issue" ?